diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/app.py ../jupyterhub-patched/jupyterhub/app.py
--- ../jupyterhub/jupyterhub/app.py	2025-10-07 16:23:32.722682432 +0200
+++ ../jupyterhub-patched/jupyterhub/app.py	2025-10-07 16:24:04.989888558 +0200
@@ -3052,6 +3052,7 @@
                     "Verifying that %s is running at %s", spawner._log_name, url
                 )
                 try:
+                    spawner.http_timeout = 10
                     await user._wait_up(spawner)
                 except AnyTimeoutError:
                     self.log.error(
