diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/app.py ../jupyterhub-patched/jupyterhub/app.py
--- ../jupyterhub/jupyterhub/app.py	2025-07-02 15:29:54.193214964 +0200
+++ ../jupyterhub-patched/jupyterhub/app.py	2025-09-10 08:12:21.799626813 +0200
@@ -3048,6 +3048,7 @@
                     "Verifying that %s is running at %s", spawner._log_name, url
                 )
                 try:
+                    spawner.http_timeout = 10
                     await user._wait_up(spawner)
                 except AnyTimeoutError:
                     self.log.error(
