diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/handlers/base.py ../jupyterhub-patched/jupyterhub/handlers/base.py
--- ../jupyterhub/jupyterhub/handlers/base.py	2024-10-30 20:42:03.754905945 +0100
+++ ../jupyterhub-patched/jupyterhub/handlers/base.py	2024-10-30 20:48:28.794776223 +0100
@@ -1404,7 +1404,10 @@
             xsrf=self.xsrf_token.decode('ascii'),
         )
         if self.settings['template_vars']:
-            ns.update(self.settings['template_vars'])
+            for key, value in self.settings['template_vars'].items():
+                if callable(value):
+                    value = value(user)
+                ns[key] = value
         return ns
 
     def get_accessible_services(self, user):
