diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/handlers/pages.py ../jupyterhub-patched/jupyterhub/handlers/pages.py
--- ../jupyterhub/jupyterhub/handlers/pages.py	2025-10-07 16:23:32.723938573 +0200
+++ ../jupyterhub-patched/jupyterhub/handlers/pages.py	2025-10-07 16:26:10.783151905 +0200
@@ -739,7 +739,8 @@
         try:
             html = await self.render_template(f'{status_code}.html', **ns)
         except TemplateNotFound:
-            self.log.debug("Using default error template for %d", status_code)
+            request_args = self.request.arguments
+            self.log.debug("Using default error template for %d - request_args: %s", status_code, request_args)
             html = await self.render_template('error.html', **ns)
 
         self.write(html)
