diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/handlers/pages.py ../jupyterhub-patched/jupyterhub/handlers/pages.py
--- ../jupyterhub/jupyterhub/handlers/pages.py	2026-03-30 15:20:37.226420874 +0200
+++ ../jupyterhub-patched/jupyterhub/handlers/pages.py	2026-03-30 15:23:34.361135858 +0200
@@ -752,7 +752,8 @@
         try:
             html = await self.render_template(f'{status_code}.html', **ns)
         except TemplateNotFound:
-            self.log.debug("Using default error template for %d", status_code)
+            request_args = self.request.arguments
+            self.log.debug("Using default error template for %d - request_args: %s", status_code, request_args)
             html = await self.render_template('error.html', **ns)
 
         self.write(html)
