diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/app.py ../jupyterhub-patched/jupyterhub/app.py
--- ../jupyterhub/jupyterhub/app.py	2025-11-27 09:28:57.845655773 +0100
+++ ../jupyterhub-patched/jupyterhub/app.py	2025-11-27 17:16:56.555135441 +0100
@@ -32,7 +32,7 @@
 from jinja2 import ChoiceLoader, Environment, FileSystemLoader, PrefixLoader
 from jupyter_events.logger import EventLogger
 from sqlalchemy.exc import OperationalError, SQLAlchemyError
-from sqlalchemy.orm import contains_eager, selectinload
+from sqlalchemy.orm import joinedload
 from tornado import gen, web
 from tornado.httpclient import AsyncHTTPClient
 from tornado.ioloop import IOLoop, PeriodicCallback
@@ -3102,10 +3102,9 @@
             .filter(orm.Spawner.server != None)
             # pre-load relationships to avoid O(N active servers) queries
             .options(
-                contains_eager(orm.User._orm_spawners),
-                selectinload(orm.Spawner.server),
+                joinedload(orm.User._orm_spawners),
+                joinedload(orm.Spawner.server),
             )
-            .populate_existing()
         ):
             # instantiate Spawner wrapper and check if it's still alive
             # spawner should be running
