diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/handlers/pages.py ../jupyterhub-patched/jupyterhub/handlers/pages.py
--- ../jupyterhub/jupyterhub/handlers/pages.py	2025-07-02 15:29:54.195214978 +0200
+++ ../jupyterhub-patched/jupyterhub/handlers/pages.py	2025-10-01 15:32:51.489288951 +0200
@@ -698,7 +698,8 @@
         try:
             html = await self.render_template(f'{status_code}.html', **ns)
         except TemplateNotFound:
-            self.log.debug("Using default error template for %d", status_code)
+            request_args = self.request.arguments
+            self.log.debug("Using default error template for %d - request_args: %s", status_code, request_args)
             html = await self.render_template('error.html', **ns)
 
         self.write(html)
