diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/user.py ../jupyterhub-patched/jupyterhub/user.py
--- ../jupyterhub/jupyterhub/user.py	2022-08-31 08:41:09.814514528 +0200
+++ ../jupyterhub-patched/jupyterhub/user.py	2022-08-31 08:43:35.714514605 +0200
@@ -783,6 +783,9 @@
                     ),
                     exc_info=True,
                 )
+            # store state again for spawner.events
+            spawner.orm_spawner.state = spawner.get_state()
+            self.db.commit()
             # raise original exception
             spawner._start_pending = False
             raise e
@@ -831,6 +834,9 @@
                     )
                 )
                 self.settings['statsd'].incr('spawner.failure.http_error')
+            # store state again for spawner.events
+            spawner.orm_spawner.state = spawner.get_state()
+            self.db.commit()
             try:
                 await self.stop(spawner.name)
             except Exception:
@@ -851,6 +857,9 @@
         finally:
             spawner._waiting_for_response = False
             spawner._start_pending = False
+            # store state again for spawner.events
+            spawner.orm_spawner.state = spawner.get_state()
+            self.db.commit()
         return spawner
 
     async def stop(self, server_name=''):
