diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/handlers/pages.py ../jupyterhub-patched/jupyterhub/handlers/pages.py
--- ../jupyterhub/jupyterhub/handlers/pages.py	2022-09-02 10:45:18.000000000 +0200
+++ ../jupyterhub-patched/jupyterhub/handlers/pages.py	2022-10-05 14:26:36.000000000 +0200
@@ -431,6 +431,7 @@
                 page,
                 user=user,
                 spawner=spawner,
+                additional_spawn_options=user.authenticator.custom_config.get("additional_spawn_options", {}),
                 progress_url=spawner._progress_url,
                 auth_state=auth_state,
             )
