diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/handlers/base.py ../jupyterhub-patched/jupyterhub/handlers/base.py
--- ../jupyterhub/jupyterhub/handlers/base.py	2023-03-28 09:51:37.095866543 +0200
+++ ../jupyterhub-patched/jupyterhub/handlers/base.py	2023-03-28 11:13:24.204263069 +0200
@@ -1342,6 +1342,7 @@
             base_url=self.hub.base_url,
             prefix=self.base_url,
             user=user,
+            user_count=self.authenticator.user_count,
             login_url=self.settings['login_url'],
             login_service=self.authenticator.login_service,
             logout_url=self.settings['logout_url'],
