diff -Naurx .git -x node_modules -x share -x __pycache__ ../traefik-proxy/jupyterhub_traefik_proxy/fileprovider.py ../traefik-proxy-patched/jupyterhub_traefik_proxy/fileprovider.py
--- ../traefik-proxy/jupyterhub_traefik_proxy/fileprovider.py	2025-11-06 18:26:41.312881379 +0100
+++ ../traefik-proxy-patched/jupyterhub_traefik_proxy/fileprovider.py	2025-11-05 15:16:27.692898028 +0100
@@ -181,7 +181,7 @@
             None: if there are no routes matching the given routespec
         """
         routespec = self.validate_routespec(routespec)
-        router_alias = traefik_utils.generate_alias(routespec, "router")
+        router_alias = traefik_utils.generate_alias(routespec, "router", self.traefik_alias_prefix)
         async with self.mutex:
             route = self.dynamic_config["jupyterhub"]["routes"].get(router_alias)
             if not route:
diff -Naurx .git -x node_modules -x share -x __pycache__ ../traefik-proxy/jupyterhub_traefik_proxy/kv_proxy.py ../traefik-proxy-patched/jupyterhub_traefik_proxy/kv_proxy.py
--- ../traefik-proxy/jupyterhub_traefik_proxy/kv_proxy.py	2025-11-06 18:26:41.312881379 +0100
+++ ../traefik-proxy-patched/jupyterhub_traefik_proxy/kv_proxy.py	2025-11-05 15:16:15.875791257 +0100
@@ -172,7 +172,7 @@
             None: if there are no routes matching the given routespec
         """
         routespec = self.validate_routespec(routespec)
-        router_alias = traefik_utils.generate_alias(routespec, "router")
+        router_alias = traefik_utils.generate_alias(routespec, "router", self.traefik_alias_prefix)
         route_key = self.kv_separator.join(
             [self.kv_jupyterhub_prefix, "routes", router_alias]
         )
diff -Naurx .git -x node_modules -x share -x __pycache__ ../traefik-proxy/jupyterhub_traefik_proxy/proxy.py ../traefik-proxy-patched/jupyterhub_traefik_proxy/proxy.py
--- ../traefik-proxy/jupyterhub_traefik_proxy/proxy.py	2025-11-06 18:26:41.312881379 +0100
+++ ../traefik-proxy-patched/jupyterhub_traefik_proxy/proxy.py	2025-11-06 18:25:56.955229158 +0100
@@ -202,10 +202,36 @@
         """,
     )
 
+    traefik_http_servers_transport = Unicode(
+        "jupyterhub",
+        config=True,
+        help="The name of the servers transport to use for internal SSL",
+    )
+
     provider_name = Unicode(
         help="""The provider name that Traefik expects, e.g. file, consul, etcd"""
     )
 
+    traefik_alias_prefix = Unicode(
+        "",
+        config=True,
+        help="""The alias prefix to use for traefik services.
+
+        This is used to namespace the services created by traefik,
+        to avoid conflicts with other services running in the same
+        environment.
+        """,
+    )
+    
+    traefik_enforce_host_in_rules = Unicode(
+        "",
+        config=True,
+        help="""
+        Optional configuration to enforce a specific host in all generated traefik rules.
+        Allows running multiple JupyterHub instances behind the same traefik proxy.
+        """,
+    )
+
     is_https = Bool(
         help="""Whether :attr:`.public_url` specifies an https entrypoint"""
     )
@@ -325,7 +351,7 @@
         # expected e.g. 'service' + '_' + routespec @ file
         routespec = self.validate_routespec(routespec)
         expected = (
-            traefik_utils.generate_alias(routespec, kind) + "@" + self.provider_name
+            traefik_utils.generate_alias(routespec, kind, self.traefik_alias_prefix) + "@" + self.provider_name
         )
         path = f"/api/http/{kind}s/{expected}"
         try:
@@ -364,7 +390,7 @@
             scale_factor=1.2,
             timeout=self.check_route_timeout,
         )
-        self.log.debug("Treafik route for %s: registered", routespec)
+        self.log.debug("Traefik route for %s: registered", routespec)
 
     async def _traefik_api_request(self, path):
         """Make an API request to traefik"""
@@ -551,13 +577,39 @@
                     }
                 }
             }
+        if self.app.internal_ssl:
+
+            def _resolve_path(path):
+                if os.path.isabs(path):
+                    return path
+                return url_path_join(os.getcwd(), path)
+
+            client_key = _resolve_path(
+                self.app.internal_proxy_certs["proxy-client"]['keyfile']
+            )
+            client_cert = _resolve_path(
+                self.app.internal_proxy_certs["proxy-client"]['certfile']
+            )
+            client_ca = _resolve_path(
+                self.app.internal_trust_bundles["proxy-client-ca"]
+            )
+            dynamic_config["http"]["serversTransports"] = {
+                self.traefik_http_servers_transport: {
+                    "certificates": [
+                        {
+                            "certfile": client_cert,
+                            "keyfile": client_key,
+                        }
+                    ],
+                    "rootCAs": client_ca,
+                }
+            }
 
         self.dynamic_config = deep_merge(dynamic_config, self.dynamic_config)
         if self.extra_dynamic_config:
             self.dynamic_config = deep_merge(
                 self.dynamic_config, self.extra_dynamic_config
             )
-
         await self._apply_dynamic_config(self.dynamic_config, None)
 
     def validate_routespec(self, routespec):
@@ -631,9 +683,9 @@
             (implementation-specific) and associated with the route
         """
 
-        service_alias = traefik_utils.generate_alias(routespec, "service")
-        router_alias = traefik_utils.generate_alias(routespec, "router")
-        rule = traefik_utils.generate_rule(routespec)
+        service_alias = traefik_utils.generate_alias(routespec, "service", self.traefik_alias_prefix)
+        router_alias = traefik_utils.generate_alias(routespec, "router", self.traefik_alias_prefix)
+        rule = traefik_utils.generate_rule(routespec, self.traefik_enforce_host_in_rules)
         # dynamic config to deep merge
         traefik_config = {
             "http": {
@@ -652,6 +704,8 @@
         traefik_config["http"]["services"][service_alias] = {
             "loadBalancer": {"servers": [{"url": target}], "passHostHeader": True}
         }
+        if self.app.internal_ssl:
+            traefik_config["http"]["services"][service_alias]["loadBalancer"]["serversTransport"] = self.traefik_http_servers_transport
 
         # Add the data node to a separate top-level node, so traefik doesn't see it.
         # key needs to be key-value safe (no '/')
@@ -711,8 +765,8 @@
 
         i.e. ( (["http", "routers", "router_name"], ("routes", "route_name") )
         """
-        service_alias = traefik_utils.generate_alias(routespec, "service")
-        router_alias = traefik_utils.generate_alias(routespec, "router")
+        service_alias = traefik_utils.generate_alias(routespec, "service", self.traefik_alias_prefix)
+        router_alias = traefik_utils.generate_alias(routespec, "router", self.traefik_alias_prefix)
         traefik_keys = (
             ["http", "routers", router_alias],
             ["http", "services", service_alias],
@@ -768,6 +822,9 @@
 
         all_routes = {}
         for _key, route in jupyterhub_config.get("routes", {}).items():
+            if self.traefik_alias_prefix and (not route.get("router", "").startswith(self.traefik_alias_prefix) and not route.get("service", "").startswith(self.traefik_alias_prefix)):
+                # not our route
+                continue
             all_routes[route["routespec"]] = {
                 "routespec": route["routespec"],
                 "data": route.get("data", {}),
diff -Naurx .git -x node_modules -x share -x __pycache__ ../traefik-proxy/jupyterhub_traefik_proxy/traefik_utils.py ../traefik-proxy-patched/jupyterhub_traefik_proxy/traefik_utils.py
--- ../traefik-proxy/jupyterhub_traefik_proxy/traefik_utils.py	2025-11-06 18:26:41.312881379 +0100
+++ ../traefik-proxy-patched/jupyterhub_traefik_proxy/traefik_utils.py	2025-11-05 15:37:15.111406325 +0100
@@ -21,7 +21,7 @@
         return u
 
 
-def generate_rule(routespec):
+def generate_rule(routespec, enforce_host_in_rules=""):
     """Generate a traefik routing rule for a jupyterhub routespec
 
 
@@ -60,6 +60,8 @@
 
     if host:
         rule = f"Host(`{host}`) && {path_rule}"
+    elif enforce_host_in_rules:
+        rule = f"Host(`{enforce_host_in_rules}`) && {path_rule}"
     else:
         rule = path_rule
     return rule
@@ -68,7 +70,7 @@
 _safe = set(string.ascii_letters + string.digits + "-")
 
 
-def generate_alias(routespec, kind=""):
+def generate_alias(routespec, kind="", alias_prefix=""):
     """Generate an alias for a routespec
 
     A safe string for use in key-value store keys, etc.
@@ -76,6 +78,8 @@
     alias = escapism.escape(routespec, safe=_safe)
     if kind:
         alias = f"{kind}_{alias}"
+    if alias_prefix:
+        alias = f"{alias_prefix}_{alias}"
     return alias
 
 
